documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,635 @@
++
+[source,shell]
+----
+cat > /tmp/alice.properties << EOF
+security.protocol=SASL_SSL
+ssl.truststore.location=/tmp/truststore.p12
+ssl.truststore.password=$STOREPASS
+ssl.truststore.type=PKCS12
+sasl.mechanism=OAUTHBEARER
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  oauth.refresh.token="$REFRESH_TOKEN" \
+  oauth.client.id="kafka-cli" \
+  oauth.ssl.truststore.location="/tmp/truststore.p12" \
+  oauth.ssl.truststore.password="$STOREPASS" \
+  oauth.ssl.truststore.type="PKCS12" \
+  oauth.token.endpoint.uri="https://$SSO_HOST/auth/realms/kafka-authz/protocol/openid-connect/token" ;
+sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler
+EOF
+----
++
+The `kafka-cli` public client is used for the `oauth.client.id` in the `sasl.jaas.config`.
+Since it's a public client it does not require a secret.
+The client authenticates with the refresh token that was authenticated in the previous step.
+The refresh token requests an access token behind the scenes, which is then sent to the Kafka broker for authentication.
+
+.Producing messages with authorized access
+
+Use the `team-a-client` configuration to check that you can produce messages to topics that start with `a_` or `x_`.
+
+. Write to topic `my-topic`.
++
+[source,shell]
+----
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9093 --topic my-topic \
+  --producer.config=/tmp/team-a-client.properties
+First message
+----
++
+This request returns a `Not authorized to access topics: [my-topic]` error.
++
+`team-a-client` has a `Dev Team A` role that gives it permission to perform any supported actions on topics that start with `a_`, but can only write to topics that start with `x_`.
+The topic named `my-topic` matches neither of those rules.
+
+. Write to topic `a_messages`.
++
+[source,shell]
+----
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9093 --topic a_messages \
+  --producer.config /tmp/team-a-client.properties
+First message
+Second message
+----
++
+Messages are produced to Kafka successfully.
+
+. Press CTRL-C to exit the CLI application.
+
+. Check the Kafka container log for a debug log of `Authorization GRANTED` for the request.
+
+[source,shell,subs="attributes"]
+----
+kubectl logs my-cluster-kafka-0 -f -n $NS
+----
+
+.Consuming messages with authorized access
+
+Use the `team-a-client` configuration to consume messages from topic `a_messages`.
+
+. Fetch messages from topic `a_messages`.
++
+[source,shell,subs=+quotes]
+----
+bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9093 --topic a_messages \
+  --from-beginning --consumer.config /tmp/team-a-client.properties
+----
++
+The request returns an error as the `Dev Team A` role for `team-a-client` only has access to consumer groups that have names starting with `a_`.
```

Comment:
```suggestion
The request returns an error because the `Dev Team A` role for `team-a-client` only has access to consumer groups that have names starting with `a_`.
```